nixos/tests/kernel-generic.nix

```diff
@@ -7,6 +7,89 @@
 with pkgs.lib;
 
 let
+  patchedPkgs = pkgs.extend (
+    final: prev: {
+      linuxKernel = prev.linuxKernel // {
+        kernelPackagesExtensions = prev.linuxKernel.kernelPackagesExtensions ++ [
+          (
+            finalKernelPackage: _:
+            let
+              inherit (finalKernelPackage) kernel;
+            in
```

Comment:
We could gain a lot of succintness here with `{ kernel, ... } @ finalKernelPackage: _:`.  Up to you.

Reply:
Attribute set unpacking is less lazy than an attribute set lookup from what I remember, so that could be part of it. Haven't looked too deeply into what's being done here though!